MekHQ/src/mekhq/campaign/mission/atb/AtBScenarioModifierApplicator.java

```diff
@@ -344,4 +344,12 @@ public static void applyObjective(AtBDynamicScenario scenario, Campaign campaign
             scenario.getScenarioObjectives().add(actualObjective);
         }
     }
+    
+    /**
+     * Applies an additional event, selected from only modifiers that benefit the player or do not benefit the player
+     */
+    public static void applyExtraEvent(AtBDynamicScenario scenario, Campaign campaign, boolean goodEvent) {
+        scenario.addScenarioModifier(AtBScenarioModifier.getRandomBattleModifier(scenario.getTemplate().mapParameters.getMapLocation(), 
+                (Boolean) goodEvent));
```

Comment:
Is the boxing here actually required?

Reply:
Yes, it may be null, in which case we take from the pool of all possible modifiers.